plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/DeployImageWizardModel.java

```diff
@@ -463,17 +471,36 @@ public void setDockerConnection(IDockerConnection dockerConnection) {
 	}
 
 	@Override
-	public boolean imageExists(String image) {
-		if (dockerConnection == null || StringUtils.isBlank(image)) {
+	public boolean imageExistsLocally(String imageName) {
+		if (dockerConnection == null || StringUtils.isBlank(imageName)) {
 			return false;
 		}
-		DockerImageURI imageName = new DockerImageURI(image);
-		String repo =  imageName.getUriWithoutTag();
-		String tag = StringUtils.defaultIfBlank(imageName.getTag(),"latest");
+		DockerImageURI imageURI = new DockerImageURI(imageName);
+		String repo =  imageURI.getUriWithoutTag();
+		String tag = StringUtils.defaultIfBlank(imageURI.getTag(),"latest");
 		return dockerConnection.hasImage(repo, tag);
 	}
 
 	@Override
+	public boolean imageExistsRemotely(String imageName) {
+		try {
+			final DockerImageURI imageURI = new DockerImageURI(imageName);
+			final String repo =  imageURI.getUriWithoutTag();
+			final List<IDockerImageSearchResult> matchingImages = dockerConnection.searchImages(repo);
+			final Optional<IDockerImageSearchResult> imageMatch = matchingImages.stream()
```

Comment:
I think it might probably need to check if the tag matches too, if it's defined in imageName